### PR TITLE
Fixed market issue and added header doc

### LINKED ISSUE #18: Market Crash
Market crashes when you buy more robotics than are available

### PR DESCRIPTION
**Pull request by Mart500**
Fixed issue with Market crashing when both the requested amount was 0 and the resource amount was 0

closes #18

### CODE REVIEW

File: core/src/io/github/teamfractal/entity/Market.java [hunk 1]
```diff
@@ -205,7 +198,15 @@ void setResource(ResourceType type, int amount)
     boolean hasEnoughResources(ResourceType type, int amount)
             throws InvalidResourceTypeException {
         int resource = getResource(type);
-        return amount <= resource;
+        /**
+         * Added by JBT to fix crash when resource and amount = 0
```
Comment (RobinStephenson) [inline core/src/io/github/teamfractal/entity/Market.java]: /** */ is reserved for docstrings. // is for a single line comment. /* */ for multi line comments

File: test/src/io/github/teamfractal/entity/MarketTest.java [hunk 2]
```diff
@@ -103,6 +97,48 @@ public void marketCanCheckResourceMoreThanAmountYouWantToBuy() {
...
+	}
+
+	/**
+	 * Made by JBT
+	 * test: hasEnoughResources
+	 * player class can use this method to find out that the amount of resource
+	 * player want to buy is available in the market, if the amount of resource
+	 * in the market is 0 then return false
+	 */
+
```
Comment (RobinStephenson) [inline test/src/io/github/teamfractal/entity/MarketTest.java]: need to remove this line for the doc generator to work

File: test/src/io/github/teamfractal/entity/MarketTest.java [hunk 3]
```diff
@@ -103,6 +97,48 @@ public void marketCanCheckResourceMoreThanAmountYouWantToBuy() {
 		assertFalse(market.hasEnoughResources(ResourceType.FOOD, 1000000));
 	}
 
+	/**
+	 * Made by JBT
```
Comment (RobinStephenson) [inline test/src/io/github/teamfractal/entity/MarketTest.java]: put the Made by JBT in their own comment above the docstring as theyre not documentation.
// Made by JBT
/**
 * docs here
 */
test here

**Review by RobinStephenson — CHANGES_REQUESTED** [review]

**Comment by Mart500** [comment]
@RobinStephenson Fixed issues

**Review by RobinStephenson — APPROVED** [review]

**Pull request merged**